src/kv/view_containers.h

```diff
@@ -3,13 +3,21 @@
 #pragma once
 
 #include "kv_types.h"
-#include "map.h"
 
 #include <functional>
 #include <map>
 
 namespace kv
 {
+  struct MapView
+  {
+    // Weak pointer to source map
+    AbstractMap* map;
+
+    // Owning pointer of TxView over that map
+    std::unique_ptr<AbstractTxView> view;
+  };
+
   // When a collection of Maps are locked, the locks must be acquired in a
   // stable order to avoid deadlocks. This ordered map will claim in name-order
   using OrderedViews = std::map<std::string, MapView>;
```

Comment:
Is `get_maps_grouped_by_domain` defined twice (in `store.h` as well).

Reply:
Yes it was, with subtly different types. I've removed it, now only `get_views_grouped_by_domain` in `tx_view.h`